dookug-common/dookug-common-cdi/src/main/java/hu/icellmobilsoft/dookug/common/cdi/template/TemplateCache.java

```diff
@@ -0,0 +1,205 @@
+/*-
+ * #%L
+ * DookuG
+ * %%
+ * Copyright (C) 2023 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.dookug.common.cdi.template;
+
+import java.security.InvalidParameterException;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import hu.icellmobilsoft.coffee.cdi.logger.AppLogger;
+import hu.icellmobilsoft.coffee.cdi.logger.ThisLogger;
+import hu.icellmobilsoft.dookug.api.dto.constants.ConfigKeys;
+import hu.icellmobilsoft.dookug.common.cdi.template.dto.TemplateCacheAction;
+import hu.icellmobilsoft.dookug.common.cdi.template.dto.TemplateCacheItem;
+
+/**
+ * Template cache functions
+ *
+ * @author istvan.peli
+ * @since 0.5.0
+ */
+@ApplicationScoped
+public class TemplateCache {
+    private final Map<String, TemplateCacheItem> cachedTemplate = new HashMap<>();
+
+    private static TemplateCacheItem firstExpiringCacheItem = null;
+
+    @Inject
+    @ThisLogger
+    private AppLogger log;
+
+    @Inject
+    @ConfigProperty(name = ConfigKeys.Cache.DOOKUG_SERVICE_TEMPLATE_MEMORY_CACHE_TTL,
+            defaultValue = ConfigKeys.Cache.DEFAULT_DOOKUG_SERVICE_TEMPLATE_MEMORY_CACHE_TTL_IN_MINUTES)
+    private Integer cacheTTLInMinutes;
+
+    /**
+     * Add a new {@link TemplateCacheItem} to the Map
+     *
+     * @param {@link
+     *            TemplateCacheItem} added element
+     */
+    public void newTemplateCacheItem(TemplateCacheItem templateCacheItem) {
+        if (templateCacheItem != null && StringUtils.isNotBlank(templateCacheItem.getTemplateId())) {
```

Comment:
Ennek hasznos lenne egy else ág is, illetve szerintem tisztázni kellene hogy mi van ha itt nullok vannak? Ha van ilyen, akkor esetleg nem-e kellene exceptiont dobni?

Reply:
else ág kész